### PR TITLE
Add '-Wextra' to compiler flags

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

This PR adds `-Wextra` to the compiler flags. It adds `-Wno-unused-parameters` because there a bunch of them related to `src/ble` and `src/platform`. It also adds `-Wno-string-concatenation` to the `mbedtls` cflags since there are some occurrences of them (fixed in recent version of mbedtls, but not in the one we use).

 #### Summary of Changes
 * Add `-Wextra` to the compiler flags

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]
A bunch of the PR description should actually be in the commit message...

**Review by rwalker-apple — APPROVED** [review]

**Review by jelderton — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@vivien-apple Looks like the EFR32 failure is real:
```

../../../examples/lock-app/efr32/third_party/connectedhomeip/examples/common/QRCode/repo/c/qrcodegen.c:208:11: error: comparison is always true due to limited range of data type [-Werror=type-limits]
  208 |  assert(0 <= (int)ecl && (int)ecl <= 3 && -1 <= (int)mask && (int)mask <= 7);
      |           ^~
...

../../../examples/lock-app/efr32/third_party/connectedhomeip/examples/common/QRCode/repo/c/qrcodegen.c:296:11: error: comparison is always true due to limited range of data type [-Werror=type-limits]
  296 |  assert(0 <= (int)ecl && (int)ecl < 4 && qrcodegen_VERSION_MIN <= version && version <= qrcodegen_VERSION_MAX);
      |           ^~
```
which kinda looks like those `(int)` casts are getting ignored for purposes of the warning, probably because the compiler assumes the incoming value is one of the defined values for that enum, and all those are non-negative.

**Review by andy31415 — APPROVED** [review]

**Comment by vivien-apple** [comment]
> @vivien-apple Looks like the EFR32 failure is real:
> 
> ```
> 
> ../../../examples/lock-app/efr32/third_party/connectedhomeip/examples/common/QRCode/repo/c/qrcodegen.c:208:11: error: comparison is always true due to limited range of data type [-Werror=type-limits]
>   208 |  assert(0 <= (int)ecl && (int)ecl <= 3 && -1 <= (int)mask && (int)mask <= 7);
>       |           ^~
> ...
> 
> ../../../examples/lock-app/efr32/third_party/connectedhomeip/examples/common/QRCode/repo/c/qrcodegen.c:296:11: error: comparison is always true due to limited range of data type [-Werror=type-limits]
>   296 |  assert(0 <= (int)ecl && (int)ecl < 4 && qrcodegen_VERSION_MIN <= version && version <= qrcodegen_VERSION_MAX);
>       |           ^~
> ```
> 
> which kinda looks like those `(int)` casts are getting ignored for purposes of the warning, probably because the compiler assumes the incoming value is one of the defined values for that enum, and all those are non-negative.

I end up adding `-Wno-type-limits` for `QRCode` only.
It brings up a few failures in `src/platform/EFR32` related `old-style-declaration` where the storage specific is not the first thing, in this case `volatile static` instead of `static volatile`

File: examples/common/QRCode/BUILD.gn [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+config("qrcode-common_config") {
+  include_dirs = [ "repo/c" ]
+
+  cflags = [ "-Wno-type-limits" ]
```
Comment (mspang) [inline examples/common/QRCode/BUILD.gn:18]: I think you should just set this globally unless we're going to fix it soon. It's pretty confusing that adding a dependency on qrcode relaxes warning flags. It means that some subset of the project including all the examples will not be bound by this warning...

**Review by mspang — APPROVED** [review]

**Pull request merged**